src/disk_io_thread_pool.cpp

```diff
@@ -186,9 +186,12 @@ namespace libtorrent
 		if (m_threads.size() == 0) return;
 		m_idle_timer.expires_from_now(reap_idle_threads_interval);
 		m_idle_timer.async_wait([this](error_code const& e) { reap_idle_threads(e); });
```

Comment:
I'm curious about the use of lambda here instead of std::bind. std::bind would be more concise, is the lambda more efficient? (I have a feeling it might be). I get the impression that the consensus in the C++ community is leaning towards using lambdas instead of std::bind.